### PR TITLE
Feature/app mesh vg health checks

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: plugins/aws/appmesh/appmeshVGHealthChecks.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        'Unable to query for App Mesh virtual gateways: ' + helpers.addError(listVirtualGateways),
+                        region, resource);
+                    continue;
+                }
+
+                if (!listVirtualGateways.data.virtualGateways) {
+                    helpers.addResult(results, 0,
+                        'No App Mesh virtual gateways found', region, resource);
+                    continue;
+                }
```
Comment (mehakseedat63) [inline plugins/aws/appmesh/appmeshVGHealthChecks.js:55]:
```suggestion
                if (!listVirtualGateways.data.virtualGateways || !listVirtualGateways.data.virtualGateways.length) {
                    helpers.addResult(results, 0,
                        'No App Mesh virtual gateways found', region, resource);
                    continue;
                }
```

virtualGateways has an empty array if there are no gateways

File: plugins/aws/appmesh/appmeshVGHealthChecks.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+                    if (describeVirtualGateway.data.virtualGateway &&
+                        describeVirtualGateway.data.virtualGateway.spec &&
+                        describeVirtualGateway.data.virtualGateway.spec.listeners) {
+                        const listeners = describeVirtualGateway.data.virtualGateway.spec.listeners;
+
+                        const hasHealthCheckPolicies = listeners.every(listener => {
+                            return listener.healthCheck && listener.healthCheck.protocol &&
+                                listener.healthCheck.healthyThreshold && listener.healthCheck.intervalMillis &&
+                                listener.healthCheck.timeoutMillis && listener.healthCheck.unhealthyThreshold;
```
Comment (mehakseedat63) [inline plugins/aws/appmesh/appmeshVGHealthChecks.js:81]:
```suggestion
                            return listener.healthCheck && Object.keys(listener.healthCheck).length;
```

maybe we should just check if healthCheck exists and not each field specifically

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/aws/appmesh/appmeshVGHealthChecks.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+                    var describeVirtualGateway = helpers.addSource(cache, source,
+                        ['appmesh', 'describeVirtualGateway', region, gateway.virtualGatewayName]);
+
+                    if (!describeVirtualGateway ||
+                        describeVirtualGateway.err ||
+                        !describeVirtualGateway.data ||
+                        !describeVirtualGateway.data.virtualGateway ||
+                        !describeVirtualGateway.data.virtualGateway.spec ||
+                        !describeVirtualGateway.data.virtualGateway.spec.listeners) {
```
Comment (AkhtarAmir) [inline plugins/aws/appmesh/appmeshVGHealthChecks.js:70]: We should have a separate check for length of listeners and if length is 0 then return a passing result

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/appmesh/appmeshVGHealthChecks.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                        helpers.addResult(results, 3,
+                            'Unable to describe App Mesh virtual gateway: ' + helpers.addError(describeVirtualGateway), region, gatewayResource);
+                        continue;
```
Comment (mehakseedat63) [inline plugins/aws/appmesh/appmeshVGHealthChecks.js:73]: There should be a separate check to see if there are no listeners, and it should pass with 'No listeners found'

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**